### ISSUE TITLE
Incremental compilation with computed property conflicting with type parameter causes spurious error

### ISSUE THREAD

**Issue by webstrand** [microsoft/TypeScript#49562]
# Bug Report

### 🔎 Search Terms

incremental compilation computed property reference type parameter

### 🕗 Version & Regression Information

- This changed between versions 3.8.3 and 3.9.7

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=3.9.7&ssl=6&ssc=37&pln=1&pc=1#code/CYUwxgNghgTiAEYD2A7AzgF3geQFzwFcUBLARwITQE8BbAIyQgG4AoUSWBDtNeAIVgAebAD54Abxbx4AbWwBdfCgL0QMaQHoN8AIKIkNAA4EMIYPEMwkhtRirwUUGlygoUSLHABmakCjAIUPB2NhawTiCm6l5WNPDEGLzIKBhQxCQoAObBVDYAdCwAviwsWvAAIgCiADKVACqV8HUAEgCSAMrw1a0Aco112E0ASq0A4qOVQ-CTQ9hDQA)

### 💻 Code

```ts
declare const O: unique symbol;
declare class Bar<O> {
  [O]: number  // A computed property name cannot reference a type parameter from its containing type.
}

// DELETE THIS LINE TO TRIGGER ERROR
```

The playground exhibits the issue, as does `tsc --watch`.

### 🙁 Actual behavior

After deleting the comment line and triggering incremental compilation, the error message

>  src/test.ts:4:4 - error TS2467: A computed property name cannot reference a type parameter from its containing type.
> 
> 3   [O]: number  // A computed property name cannot reference a type parameter from its containing type.

appears.

### 🙂 Expected behavior

No error should appear after deleting the comment or triggering incremental compilation.
